From 138af3b728f823bc745eb4a63a61dddc35996b42 Mon Sep 17 00:00:00 2001
From: Jiuyang Liu <liu@jiuyang.me>
Date: Tue, 26 Jul 2022 08:53:08 +0000
Subject: [PATCH] fix for cde 1.2

---
 src/main/scala/package.scala | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/src/main/scala/package.scala b/src/main/scala/package.scala
index 7a634b620af..53dea6b3ff4 100644
--- a/src/main/scala/package.scala
+++ b/src/main/scala/package.scala
@@ -1,7 +1,12 @@
 // See LICENSE.SiFive for license details.
 
-package freechips
-
-package object rocketchip {
-  val config = chipsalliance.rocketchip.config
+package freechips {
+  package object rocketchip {
+    val config = org.chipsalliance.cde.config
+  }
+}
+package chipsalliance {
+  package object rocketchip {
+    val config = org.chipsalliance.cde.config
+  }
 }
